### PR TITLE
Mobile: Allow for camel case keys in appointment reason code embedded data

### PR DESCRIPTION
**Pull request by aherzberg**
## Summary
In some cases, the acheron embedded, reason code comes in as camel case, causing mobile to not get the value. This allows for both normal spaced case and camel case.  

## Related issue(s)
https://github.com/department-of-veterans-affairs/va-mobile-app/issues/7225

## Testing done
Specs 

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
Mobile

## Acceptance criteria

- [ ]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: modules/mobile/app/models/mobile/v0/adapters/vaos_v2_appointment.rb [hunk 1]
```diff
@@ -475,12 +475,18 @@ def embedded_data
         end
 
         def embedded_data_match(key)
-          match = appointment.dig(:reason_code, :text)&.match(/(^|\|)#{key}:?(.*?)(\||$)/)
+          camelized_key = key.gsub(' ', '_').camelize(:lower)
```
Comment (kpethtel) [inline modules/mobile/app/models/mobile/v0/adapters/vaos_v2_appointment.rb:478]: So you're telling me that they're sometimes giving us keys like `preferred dates` and sometimes `preferredDates`? That seems ridiculously sloppy, like they're setting us up for failure.

File: modules/mobile/spec/models/adapters/appointments_vaos_v2_adapter_spec.rb [hunk 2]
```diff
@@ -913,5 +913,22 @@ def appointment_data(index = nil)
         expect(result.reason).to be_nil
       end
     end
+
+    context 'when some acheron field keys are camel case' do
+      it 'parses both camel case and non camel case fields' do
+        appointment = appointment_data[12]
+        appointment[:reason_code][:coding] = [{ code: 'will not be used' }]
```
Comment (kpethtel) [inline modules/mobile/spec/models/adapters/appointments_vaos_v2_adapter_spec.rb:920]: It took me a few minutes to understand that these values are essentially holdovers from the above test to confirm that the embedded values are used preferentially over these values.

**Review by kpethtel — APPROVED** [review]
Frustrating that this is how the upstream works, but your changes look good.

File: modules/mobile/app/models/mobile/v0/adapters/vaos_v2_appointment.rb (hunk 1 above)
Comment (aherzberg) [reply modules/mobile/app/models/mobile/v0/adapters/vaos_v2_appointment.rb:478]: I've only seen this happen with the "reason code" vs reasonCode but I figured it wouldn't be impossible for it to happen to other fields as well so wanted to code defensively for it.

Comment (kpethtel) [reply modules/mobile/app/models/mobile/v0/adapters/vaos_v2_appointment.rb:478]: Does this PR fix the reason code issue? That's not apparent to me.

Comment (aherzberg) [reply modules/mobile/app/models/mobile/v0/adapters/vaos_v2_appointment.rb:478]: Yeah, it goes through all the embedded data fields, one of which is the reason code and checks for normal spaced case or camel case as a fall back. so no change in behavior if the original field is there.

Comment (kpethtel) [reply modules/mobile/app/models/mobile/v0/adapters/vaos_v2_appointment.rb:478]: I see. So when you say reason_code, you mean:
`reason_code[:text][:reason_code]`
not the top level reason code.

**Pull request merged**